ibm/resource_ibm_private_dns_glb_pool.go

```diff
@@ -0,0 +1,342 @@
+			},
+			pdnsGlbPoolModifiedOn: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceIBMPrivateDNSGLBPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	instanceID := d.Get(pdnsInstanceID).(string)
+	CreatePoolOptions := sess.NewCreatePoolOptions(instanceID)
+
+	poolname := d.Get(pdnsGlbPoolName).(string)
+	CreatePoolOptions.SetName(poolname)
+
+	if description, ok := d.GetOk(pdnsGlbPoolDescription); ok {
+		CreatePoolOptions.SetDescription(description.(string))
+	}
+	if enable, ok := d.GetOk(pdnsGlbPoolEnabled); ok {
+		CreatePoolOptions.SetEnabled(enable.(bool))
+	}
+	if threshold, ok := d.GetOk(pdnsGlbPoolHealthyOriginsThreshold); ok {
+		CreatePoolOptions.SetHealthyOriginsThreshold(int64(threshold.(int)))
+	}
+	if monitor, ok := d.GetOk(pdnsGlbPoolMonitor); ok {
+		CreatePoolOptions.SetMonitor(monitor.(string))
+	}
+	if chanel, ok := d.GetOk(pdnsGlbPoolChannel); ok {
+		CreatePoolOptions.SetNotificationChannel(chanel.(string))
+	}
+	if region, ok := d.GetOk(pdnsGlbPoolRegion); ok {
+		CreatePoolOptions.SetHealthcheckRegion(region.(string))
+	}
+	if _, ok := d.GetOk(pdnsGlbPoolSubnet); ok {
+		CreatePoolOptions.SetHealthcheckSubnets(expandStringList(d.Get(pdnsGlbPoolSubnet).([]interface{})))
+	}
+
+	poolorigins := d.Get(pdnsGlbPoolOrigins).(*schema.Set)
+	CreatePoolOptions.SetOrigins(expandPoolOrigins(poolorigins))
+
+	result, resp, err := sess.CreatePool(CreatePoolOptions)
+	if err != nil {
+		log.Printf("create global load balancer pool failed %s", resp)
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", instanceID, *result.ID))
+	d.Set(pdnsGlbPoolID, *result.ID)
+
+	return resourceIBMPrivateDNSGLBPoolRead(d, meta)
+}
+
+func resourceIBMPrivateDNSGLBPoolRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+	idset := strings.Split(d.Id(), "/")
+
+	getPoolOptions := sess.NewGetPoolOptions(idset[0], idset[1])
+	presponse, resp, err := sess.GetPool(getPoolOptions)
+	if err != nil {
+		return fmt.Errorf("Error fetching pdns GLB Pool:%s\n%s", err, resp)
+	}
+
+	response := *presponse
+	d.Set(pdnsGlbPoolName, response.Name)
+	d.Set(pdnsGlbPoolID, response.ID)
+	d.Set(pdnsGlbPoolDescription, response.Description)
+	d.Set(pdnsGlbPoolEnabled, response.Enabled)
+	d.Set(pdnsGlbPoolHealthyOriginsThreshold, response.HealthyOriginsThreshold)
+	d.Set(pdnsGlbPoolMonitor, response.Monitor)
+	d.Set(pdnsGlbPoolChannel, response.NotificationChannel)
+	d.Set(pdnsGlbPoolRegion, response.HealthcheckRegion)
+	d.Set(pdnsGlbPoolSubnet, response.HealthcheckSubnets)
+	d.Set(pdnsGlbPoolCreatedOn, response.CreatedOn)
+	d.Set(pdnsGlbPoolModifiedOn, response.ModifiedOn)
+	d.Set(pdnsGlbPoolOrigins, flattenPoolOrigins(response.Origins))
+
+	log.Printf("global load balancer pool created successfully : %s", *presponse.ID)
```

Comment:
This is generic read.. This read function ll called for update import as well..
So You can remove the log statement or either make the print statement generic

Reply:
done